game-app/game-core/src/main/java/games/strategy/triplea/ui/BattleDisplay.java

```diff
@@ -775,7 +782,14 @@ void setNotificationShort(
 
     private void categorizeUnits(
         final Iterable<UnitCategory> categoryIter, final boolean damaged, final boolean disabled) {
-      for (final UnitCategory category : categoryIter) {
+      final List<UnitCategory> unitCategories =
+          StreamSupport.stream(categoryIter.spliterator(), false).collect(Collectors.toList());
+      if (unitCategories.isEmpty()) {
+        return;
+      }
+      final GameData gameData = unitCategories.get(0).getUnitAttachment().getData();
```

Comment:
`BattleDisplay:L95` has a reference to `GameData` - can that be used?

Reply:
It cannot as the changed code is in private static class `CasualtyNotificationPanel` for which I could not find `GameData`.

Reply:
NP - please double check the nullability of `unitAttachment#data`. I think grabbing gameData like this should be 'safe', but let's be sure.

Reply:
What should happen if `unitAttachment = null` ? For the sorting a different gameData source would be needed, e.g. from `unitAttachment#unitType`? If all potential sources could be null, we cannot escape this what-if-question.

Reply:
I looked into this a bit further. You're good here. I'll post some more followup in a bit.

Reply:
Correction.. tracing this down... Neutral units in 2.5 have null `gameData`

If you save a game that has neutral players with 2.5, then load that up with 2.6, then find a territory with only neutral units - in that case `gameData` would seemingly be null. 

Happily that nullity concern will go away kinda soon as how the neutral player is saved to disk was changed. Once we no longer need to load saves from 2.5, that `gameData` would never be null.